client/lib/wpcom-undocumented/lib/undocumented.js

```diff
@@ -2350,6 +2350,34 @@ Undocumented.prototype.transferStatus = function( siteId, transferId ) {
 };
 
 /**
+ * Update the poster for a video.
+ *
+ * @param {string} videoId ID of the video
+ * @param {object} data The POST data
+ * @param {Function} fn Function to invoke when request is complete
+ * @returns {Promise} A promise that resolves when the request is complete
+ */
+Undocumented.prototype.updateVideoPoster = function( videoId, data, fn ) {
+	debug( '/videos/:video_id/poster' );
+
+	const params = {
+		path: `/videos/${ videoId }/poster`,
+	};
+
+	if ( 'file' in data ) {
+		params.formData = [
+			[ 'poster', data.file ]
+		];
+	}
+
+	if ( 'at_time' in data ) {
```

Comment:
I would make this with an `elseif`, since you can't send both (the logic will always take the `at_time` first on the server side). Also, you should have an `else` at the end, that if neither is present, it will cause this to not post to the API.